app/controllers/users_controller.rb

```diff
@@ -1,5 +1,25 @@
 class UsersController < ApplicationController
   def show
-    @user = User.find_by(username: params[:username])
+      @user = User.find_by(username: params[:username])
+      redirect_to posts_path, alert: "Could not find a user with that name" unless @user
+  end
+
+  def follow
+    other_user = params[:user]
+
+    if current_user.is_following?(other_user)
+      notice = "You will continue following this user"
+    else
+      Relationship.create(follower_id: current_user.id, followed_id: other_user)
+      notice = "You are now following this user"
+    end
+    redirect_to profile_path(User.find(other_user).username), notice: notice
+  end
+
+  def unfollow
+    followed = params[:user]
```

Comment:
Same here. I'd convert it to a `User` on this line, to make it easier to use below.